orbiter_translations/data_stage/xml_demo.py

```diff
@@ -0,0 +1,168 @@
+    dag_filter_rule,
+    dag_rule,
+    task_filter_rule,
+    task_rule,
+    task_dependency_rule,
+    post_processing_rule,
+    cannot_map_rule,
+)
+from orbiter.rules.rulesets import (
+    DAGFilterRuleset,
+    DAGRuleset,
+    TaskFilterRuleset,
+    TaskRuleset,
+    TaskDependencyRuleset,
+    PostProcessingRuleset,
+    TranslationRuleset,
+)
+
+
+@dag_filter_rule
+def basic_dag_filter(val: dict) -> list | None:
+    """Filter input down to a list of dictionaries that can be processed by the `@dag_rules`"""
+    return val["DSExport"][0]["Job"]
+
+
+@dag_rule
+def basic_dag_rule(val: dict) -> OrbiterDAG | None:
+    """Translate input into an `OrbiterDAG`"""
+    try:
+        dag_id = val["@Identifier"]
+        dag_id = inflection.underscore(dag_id)
+        return OrbiterDAG(dag_id=dag_id, file_path=f"{dag_id}.py")
+    except Exception:
+        return None
+
+
+@task_filter_rule
+def basic_task_filter(val: dict) -> list | None:
+    """Filter input down to a list of dictionaries that can be processed by the `@task_rules`"""
+    if isinstance(val, dict):
+        val = json.loads(json.dumps(val, default=str))  # pre-serialize values, for JQ
+        try:
+            return (
+                jq.compile(""".Record[] | select(.["@Type"] == "CustomStage")""")
+                .input_value(val)
+                .all()
+            )
+        except StopIteration:
+            pass
+    return None
+
+
+@task_rule(priority=2)
+def basic_task_rule(val: dict) -> OrbiterOperator | OrbiterTaskGroup | None:
+    """Translate input into an Operator (e.g. `OrbiterBashOperator`). will be applied first, with a higher priority"""
+    if "task_id" in val:
+        return OrbiterEmptyOperator(task_id=val["task_id"])
+    else:
+        return None
+
+
+def task_common_args(val: dict) -> dict:
+    """
+    Common mappings for all tasks
+    """
+    task_id: str = (
+        jq.compile(""".Property[] | select(.["@Name"] == "Name") | .["#text"]""")
+        .input_value(val)
+        .first()
+    )
+    task_id = inflection.underscore(task_id)
+    params = {"task_id": task_id}
+    return params
+
+
+def extract_sql_statements(root):
+    sql_statements = {}
+    sql_tags = ["SelectStatement", "BeforeSQL", "AfterSQL"]
+
+    for tag in sql_tags:
+        elements = root.findall(f".//{tag}")
+        for elem in elements:
+            if elem.text:
+                sql_text = elem.text.strip()
+                sql_statements[tag] = sql_text
+    return sql_statements
+
+
+@task_rule(priority=2)
+def sql_command_rule(val) -> OrbiterSQLExecuteQueryOperator | None:
+    """
+    For SQLQueryOperator.
+
+    """  # noqa: E501
+    try:
+        sql: str = (
+            jq.compile(
+                """.Collection[] | .SubRecord[] | .Property[] | select(.["@PreFormatted"] == "1") | .["#text"] """
+            )
+            .input_value(val)
+            .first()
+        )
+        root = ElementTree.fromstring(sql.encode("utf-16"))
```

Comment:
We are already parsing XML - IMO we should adjust the initial parser to also parse any embedded xml, rather than each rule needing to try to figure it out